topics/statistics/tutorials/intro-to-ml-with-python/tutorial.md

```diff
@@ -0,0 +1,4133 @@
+---
+layout: tutorial_hands_on
+title: Foundational Aspects of Machine Learning using Python
+level: Intermediate
+draft: true
+requirements:
+-
+  type: "internal"
+  topic_name: data-science
+  tutorials:
+  - python-basics
+  - python-warmup-stat-ml
+questions:
+- to do
+objectives:
+- general sklearn syntax intro
+- overfit/underfit
+- the need for regularization
+- cross validation and a test set
+- metrics and imbalance
+time_estimation: 3H
+key_points:
+- To be added
+contributions:
+  authorship:
+  - wandrilled
+tags:
+- elixir
+- ai-ml
+priority: 2
+notebook:
+  language: python
+  pyolite: true
+---
+
+
+```python
+%reload_ext autoreload
+%autoreload 2
+```
+
+
+```python
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import matplotlib.pylab as pylab
+import pandas as pd
+from operator import itemgetter
+
+from warnings import filterwarnings
+
+pylab.rcParams['figure.figsize'] = 5, 5
+plt.rc("font", size=10)
+
+
+plt.rc('xtick', color='k', labelsize='medium', direction='in')
+plt.rc('xtick.major', size=8, pad=12)
+plt.rc('xtick.minor', size=8, pad=12)
+
+plt.rc('ytick', color='k', labelsize='medium', direction='in')
+plt.rc('ytick.major', size=8, pad=12)
+plt.rc('ytick.minor', size=8, pad=12)
+```
+
+So far we have fitted our curves and doing so we have found the  best model explaining the point that we had for the fitting.
+
+We also saw that we could choose different models according to how much the improvement obtained was worth the complexification of the model. But again we did it on the whole data that we had. We never really check how well our data was generalizing to points never seen before, or by how much the model we found was subject to outliers.
+
+Machine learning procedures allow us to take those considerations into account. After highlighting the few caveats of the procedures we have used in the former notebook, we will introduce the foundation of the machine learning way to model.
+
+More particularly we will see that the machine learning paradigm modifies the function to optimize that we have seen before by adding a penalty to covariables that generalize badly. We will also see that in a machine learning procedure, the generalization is approached by fitting and evaluating mutliple times your model on subset of your data.
+
+**The machine learning paradigm emphasizes the importance of building a general model that will be good at dealing with future, unknown, data points rather than being the best model on the data that we have now.**
```

Comment:
```suggestion
The machine learning paradigm emphasizes the importance of building a general model that will be good at dealing with future, unknown, data points rather than being the best model on the data that we have now.
```